### PR TITLE
Update status in case Bastion blocks Shoot deletion

### LINKED ISSUE #5346: Update status in case Bastion blocks Shoot deletion
**How to categorize this issue?**

/area quality
/kind bug

**What happened**:
`Shoot` deletion is blocked in case there are still `Bastions` for this `Shoot`
`{"level":"info","msg":"Error syncing Shoot garden-core/delete-test2: shoot has still Bastions","ts":"2022-01-26T12:50:47.384Z"}`
The Info is logged but the reason is not apparent on the `Shoot` state.

https://github.com/gardener/gardener/blob/31b8385f6fbae62af46e18f9c55a6e7624cec16e/pkg/gardenlet/controller/shoot/shoot_control.go#L377-L383

**What you expected to happen**:
Maybe the `lastOperation` should be updated accordingly?

**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know?**:
related issue #5323 

**Environment**:

- Gardener version:
- Kubernetes version (use `kubectl version`):
- Cloud provider or hardware configuration:
- Others:

### PR DESCRIPTION
**Pull request by petersutter**
**How to categorize this PR?**

/area ops-productivity
/kind enhancement

**What this PR does / why we need it**:
The Shoot status is updated accordingly in case there are Bastions that block the Shoot deletion.

**Which issue(s) this PR fixes**:
Fixes #5346 

**Special notes for your reviewer**:

**Release note**:

```other operator

```

### CODE REVIEW

File: pkg/gardenlet/controller/shoot/shoot_control.go [hunk 1]
```diff
@@ -379,7 +381,9 @@ func (r *shootReconciler) deleteShoot(ctx context.Context, logger logrus.FieldLo
 		return reconcile.Result{}, fmt.Errorf("failed to check for related Bastions: %w", err)
 	}
 	if hasBastions {
-		return reconcile.Result{}, errors.New("shoot has still Bastions")
+		hasBastionError := errors.New("shoot has still Bastions")
```
Comment (ialidzhikov) [inline pkg/gardenlet/controller/shoot/shoot_control.go]: For consistency with the above similar error:
```suggestion
		hasBastionErr := errors.New("shoot has still Bastions")
```

**Review by ialidzhikov — COMMENTED** [review]
Minor nit, otherwise lgtm.

File: pkg/gardenlet/controller/shoot/shoot_control.go (hunk 1 above)
Comment (petersutter) [reply pkg/gardenlet/controller/shoot/shoot_control.go]: ✅

**Review by ialidzhikov — APPROVED** [review]
/lgtm
/squash

**Review by rfranzke — APPROVED** [review]

**Pull request merged**